mlflow/langchain/databricks_dependencies.py

```diff
@@ -47,6 +53,14 @@ def _get_vectorstore_from_retriever(retriever) -> Generator[Resource, None, None
         if isinstance(embeddings, (DatabricksEmbeddings, LegacyDatabricksEmbeddings)):
             yield DatabricksServingEndpoint(endpoint_name=embeddings.endpoint)
 
+def _extract_databricks_dependencies_from_uc_function_toolkit(uc_toolkit) -> Generator[Resource, None, None]:
+    from langchain_community.tools.databricks import UCFunctionToolkit
+
+    if isinstance(uc_toolkit, UCFunctionToolkit):
+        yield DatabricksSQLWarehouse(warehouse_id=uc_toolkit.warehouse_id)
+    
+        for function_name in uc_toolkit.tools.keys():
+            yield DatabricksUCFunction(function_name=function_name)
```

Comment:
Can we make sure that this is an instance of a string or FunctionInfo or something? So if langchain changes something this won't break

Reply:
Its a str. Looked at the source code here: https://api.python.langchain.com/en/latest/_modules/langchain_community/tools/databricks/tool.html#UCFunctionToolkit

Tools is defined as a dictionary where the key is a string:

<img width="882" alt="image" src="https://github.com/user-attachments/assets/1c20b9ec-2551-4ec0-8dda-b53d2bb24ed5">